File: quiz-platform-backend/src/main/java/com/quizplatform/service/AuthService.java
package com.quizplatform.service;

import com.quizplatform.dto.request.*;
import com.quizplatform.dto.response.AuthResponse;
import com.quizplatform.dto.response.UserResponse;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    AuthResponse refreshToken(RefreshTokenRequest request);
    void logout(String refreshToken);
    void forgotPassword(ForgotPasswordRequest request);
    void resetPassword(ResetPasswordRequest request);
    void changePassword(ChangePasswordRequest request, String email);
    UserResponse getCurrentUser(String email);
    UserResponse updateProfile(UpdateProfileRequest request, String email);
}

File: quiz-platform-frontend/src/services/authService.ts
import api from '@/utils/api';
import { 
  LoginRequest, 
  RegisterRequest, 
  AuthResponse, 
  ApiResponse, 
  ChangePasswordRequest 
} from '@/types';

export const authService = {
  login: async (data: LoginRequest) => {
    const response = await api.post<ApiResponse<AuthResponse>>('/auth/login', data);
    return response.data;
  },

  register: async (data: RegisterRequest) => {
    const response = await api.post<ApiResponse<void>>('/auth/register', data);
    return response.data;
  },

  logout: async (refreshToken: string) => {
    const response = await api.post<ApiResponse<void>>('/auth/logout', { refreshToken });
    return response.data;
  },

  refreshToken: async (refreshToken: string) => {
    const response = await api.post<ApiResponse<{ accessToken: string }>>('/auth/refresh', { refreshToken });
    return response.data;
  },

  forgotPassword: async (email: string) => {
    const response = await api.post<ApiResponse<void>>('/auth/forgot-password', { email });
    return response.data;
  },

  resetPassword: async (token: string, password: string) => {
    const response = await api.post<ApiResponse<void>>('/auth/reset-password', { token, password });
    return response.data;
  },

  changePassword: async (data: ChangePasswordRequest) => {
    const response = await api.post<ApiResponse<void>>('/auth/change-password', data);
    return response.data;
  }
};
